registry/storage/driver/factory/factory.go

```diff
@@ -37,12 +41,61 @@ func Register(name string, factory StorageDriverFactory) {
 // parameters. To use a driver, the StorageDriverFactory must first be
 // registered with the given name. If no drivers are found, an
 // InvalidStorageDriverError is returned
-func Create(name string, parameters map[string]interface{}) (storagedriver.StorageDriver, error) {
+func Create(ctx context.Context, name string, parameters map[string]interface{}) (storagedriver.StorageDriver, error) {
 	driverFactory, ok := driverFactories[name]
 	if !ok {
 		return nil, InvalidStorageDriverError{name}
 	}
-	return driverFactory.Create(parameters)
+	d, err := driverFactory.Create(parameters)
+	if err != nil {
+		return nil, err
+	}
+	return d, verify(ctx, d)
+}
+
+// Ensure that the configured storage driver has permissions for the registry to function
+// i.e. read, write and delete a file
+func verify(ctx context.Context, driver storagedriver.StorageDriver) error {
+	randomFile := fmt.Sprintf("/%s", uuid.Generate().String())
+	err := driver.PutContent(ctx, randomFile, []byte(""))
+	if err != nil {
+		return fmt.Errorf("unable to write verification file: %s", err)
```

Comment:
Maybe a higher level message, telling the user what is to be blamed? Like: `Your registry storage permissions are improperly configured. Please check your settings (error: unable to write verification file: %s)`